homeassistant/components/dobiss/__init__.py

```diff
@@ -0,0 +1,126 @@
+"""The dobiss integration."""
+
+import logging
+from typing import Any
+
+from dobissapi import DobissAPI
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryError, ConfigEntryNotReady
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema("dobiss")
+
+PLATFORMS: list[Platform] = [Platform.LIGHT]
+
+
+async def async_setup(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up the dobiss component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up dobiss from a config entry."""
+    client = DobissClient(hass, entry)
+    entry.runtime_data = client
+
+    if not await client.async_setup():
+        _LOGGER.warning("Dobiss setup failed")
+        raise ConfigEntryError("Invalid Dobiss configuration")
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload a config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id, None)
+        if not hass.data[DOMAIN]:
+            hass.data.pop(DOMAIN)
+    else:
+        _LOGGER.warning("Unload failed")
+
+    return unload_ok
+
+
+async def async_reload_entry(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Reload config entry."""
+    await async_unload_entry(hass, entry)
+    await async_setup_entry(hass, entry)
+
+
+class DobissClient:
+    """Handle communication and setup logic for a single Dobiss config entry."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the Dobiss data."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.api = None
+        self.available = False
+        self.unsub = None
+        self.devices: list[Any] = []
+
+    @property
+    def host(self):
+        """Return client host."""
+        return self.config_entry.data[CONF_HOST]
+
+    async def async_setup(self):
+        """Set up the Dobiss client."""
+        try:
+            self.api = DobissAPI(
+                self.config_entry.data["secret"],
+                self.config_entry.data["host"],
+                self.config_entry.data["secure"],
+            )
+
+            devices = self.api.get_all_devices()
+            self.devices = devices
+
+            await self.api.discovery()
+            self.hass.async_create_task(self.api.dobiss_monitor())
+
+            self.available = True
+            _LOGGER.debug("Successfully connected to Dobiss")
```

Comment:
Only that which can raise should be in the try-clause. I'm guessing it's only `self.api.discovery()` which actually raises?